.github/workflows/build.yml

```diff
@@ -31,24 +32,24 @@ jobs:
         python-version: ["3.8", "3.9", "3.10", "3.11"]
     steps:
       - uses: actions/checkout@v3
+      - uses: pdm-project/setup-pdm@v3
```

Comment:
why is that needed? we can and should install the project with pip as this is how it will be installed by users. If there is any difference between pdm and pip, then pip is a preferred way because we need to test how it will behave when installed with pip. If there is no difference, then it is just making tests longer to execute because of an additional installation step.

Reply:
A few reasons:
- We are now using the pdm build backend.
- More concise and manageable workflows (Which could still use a few edits. I'm about to circle back to it).
- Recall that since pdm strictly conforms to the same PEPs as pip, there is no danger of difference. This is not true of Poetry, and so you definitely wouldn't do this with poetry. With pdm on the other hand, there is no downside.

The main reason I got into editing those files was because setup.py (Which we no longer have/need) was referenced in them. It was then that I realized pip was being used everywhere, and that if we're now strictly using pyproject.toml (As we should) pdm is the most sensible way of dealing with that.

Reply:
Also related to: https://github.com/microsoft/autogen/pull/1155#discussion_r1449573001

Reply:
I am sorry, I was not precise enough. My question was not about whether we should use `pdm` as a build system or not, but why don't we install `autogen` in this particular script using `pip`. More precisely, if we replace
```
pdm install -d
```
with
```
pip install -e .
```
below, then we would still use `pdm` backend but through `pip`. Since our users install autogen using `pip`, it would make CI closer to the actual use.

Of course, those two lines should behave the same, but there are always some bugs in both `pip` and `pdm` that could cause different behaviours. In such rare cases, you really want to stick with `pip install` because that way you will catch such bugs in CI.